### PR TITLE
feat: daemon: improvemens to the consensus slasher

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

Improvements to the new consensus slasher that was created in #10871 

## Proposed Changes

- Log failures
- Return a bool to indicate any faults, separate from other errors
- Perform some additional checks before reporting faults 

## Additional Info

 Run `lotus daemon --slash-consensus=true --slashdb-dir="/mnt/data/.slasher" >> /mnt/data/slash.log 2>&1 &` to run the consensus slasher.

`--slashdb-dir` is used to supply the path to the slasher database.
`--slasher-sender` can optionally be used to specify the account to send the `ReportConsensusFault` message from.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by jennijuju** [comment]
missing  bare minimum instruction on how to run this thing

**Comment by jennijuju** [comment]
> missing bare minimum instruction on how to run this thing

gentle ping

**Comment by mx819812523** [comment]
Run ```nohup lotus daemon --slash-consensus=true --slashdb-dir="/mnt/data/.slasher``` to monitor consensus slasher
```--slash-consensus``` is a flag to open monitor and it need to be used with ```-- slashdb-dir```, ```--slasher-sender```optionally specify the account to report consensus from

File: cmd/lotus/daemon.go [hunk 1]
```diff
@@ -639,10 +638,12 @@ func slashConsensus(a lapi.FullNode, p string, from string) error {
...
-			if otherBlock == nil {
-				continue
-			}
+		otherBlock, extraBlock, fault, err := slashFilterMinedBlock(ctx, sf, a, block)
+		if err != nil {
+			log.Errorf("slash detector errored: %s", err)
+			continue
+		}
+		if fault {
 			log.Errorf("<!!> SLASH FILTER ERROR: %s", err)
```
Comment (ZenGround0) [inline cmd/lotus/daemon.go:647]: I think we are losing information here.  Before this change the error contained information about the slashing condition being hit and the problem blocks.  Now I think this err will just be nil

File: miner/miner.go [hunk 2]
```diff
@@ -324,9 +324,16 @@ minerLoop:
...
+			if os.Getenv("LOTUS_MINER_NO_SLASHFILTER") != "_yes_i_know_i_can_and_probably_will_lose_all_my_fil_and_power_" {
+				witness, fault, err := m.sf.MinedBlock(ctx, b.Header, base.TipSet.Height()+base.NullRounds)
+				if err != nil {
+					log.Errorf("<!!> SLASH FILTER ERRORED: %s", err)
+					// Continue here, because it's _probably_ wiser to not submit this block
+					continue
+				}
+
+				if fault {
+					log.Errorf("<!!> SLASH FILTER DETECTED FAULT due to witness %s", witness)
```
Comment (ZenGround0) [inline miner/miner.go:336]: Again I think the old error log had more information about the slashing condition and blocks causing it.

**Review by ZenGround0 — APPROVED** [review]
I won't block the review but will advocate for maintaining logging that provides useful information.  This might mean logging when the fault is detected and the information is available or maybe going back and using typed errors instead of a bool.

With regards to "missing bare minimum instruction on how to run this thing" I have recently learned that [this](https://github.com/filecoin-project/lotus/discussions/categories/tutorials) is a good place to put such things

**Comment by mx819812523** [comment]
My previous log error: MpoolPushMessage error log is %s not %w

File: cmd/lotus/daemon.go (hunk 1 above)
Comment (arajasek) [reply cmd/lotus/daemon.go:647]: Good catch, thank you -- fixing.

**Pull request merged**